examples/mechanics/fragmenting_cylinder.cpp

```diff
@@ -0,0 +1,161 @@
+    std::array<double, 3> high_corner = inputs["high_corner"];
+    std::array<int, 3> num_cells = inputs["num_cells"];
+    int m = std::floor( delta /
+                        ( ( high_corner[0] - low_corner[0] ) / num_cells[0] ) );
+    int halo_width = m + 1; // Just to be safe.
+
+    // ====================================================
+    //                    Force model
+    // ====================================================
+    using model_type = CabanaPD::ForceModel<CabanaPD::PMB, CabanaPD::Fracture>;
+    model_type force_model( delta, K, G0 );
+    // using model_type =
+    //      CabanaPD::ForceModel<CabanaPD::LPS, CabanaPD::Fracture>;
+    // model_type force_model( delta, K, G, G0 );
+
+    // ====================================================
+    //                 Particle generation
+    // ====================================================
+    // Does not set displacements, velocities, etc.
+    auto particles = std::make_shared<
+        CabanaPD::Particles<memory_space, typename model_type::base_model,
+                            typename model_type::thermal_type>>(
+        exec_space(), low_corner, high_corner, num_cells, halo_width );
+
+    // ====================================================
+    //            Custom particle initialization
+    // ====================================================
+    double x_center = 0.5 * ( low_corner[0] + high_corner[0] );
+    double y_center = 0.5 * ( low_corner[1] + high_corner[1] );
+    double Rout = inputs["cylinder_outer_radius"];
+    double Rin = inputs["cylinder_inner_radius"];
+
+    // Do not create particles outside given cylindrical region
+    auto init_op = KOKKOS_LAMBDA( const int, const double x[3] )
+    {
+        double rsq = ( x[0] - x_center ) * ( x[0] - x_center ) +
+                     ( x[1] - y_center ) * ( x[1] - y_center );
+        if ( rsq < Rin * Rin || rsq > Rout * Rout )
+            return false;
+        return true;
+    };
+    particles->createParticles( exec_space(), init_op );
+
+    auto rho = particles->sliceDensity();
+    auto x = particles->sliceReferencePosition();
+    auto v = particles->sliceVelocity();
+    auto f = particles->sliceForce();
+
+    double vrmax = inputs["max_radial_velocity"];
+    double vrmin = inputs["min_radial_velocity"];
+    double vzmax = inputs["max_vertical_velocity"];
+    double zmin = low_corner[2];
+
+    double dx = particles->dx[0];
+    double dy = particles->dx[1];
+    double dz = particles->dx[2];
+
+    double height = inputs["system_size"][2];
+    auto init_functor = KOKKOS_LAMBDA( const int pid )
+    {
+        // Density
+        rho( pid ) = rho0;
+
+        // Perturb particle positions
+        double random_number_x = ( (double)rand() / ( RAND_MAX ) );
+        double random_number_y = ( (double)rand() / ( RAND_MAX ) );
+        double random_number_z = ( (double)rand() / ( RAND_MAX ) );
+
+        double factor = 0.001; // inputs["grid_perturbation_factor"];
```

Comment:
When trying to use 

double factor = inputs["grid_perturbation_factor"];

I got the error below. Can you fix this?

CabanaPD/examples/mechanics/fragmenting_cylinder.cpp:123:31: error: no viable overloaded operator[] for type 'const CabanaPD::Inputs'
        double factor = inputs["grid_perturbation_factor"];
                        ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
CabanaPD/src/CabanaPD_Input.hpp:185:10: note: candidate function not viable: 'this' argument has type 'const CabanaPD::Inputs', but method is not marked const
    auto operator[]( std::string label ) { return inputs[label]["value"]; }
         ^
1 error generated.
make[2]: *** [examples/mechanics/CMakeFiles/FragmentingCylinder.dir/fragmenting_cylinder.cpp.o] Error 1
make[1]: *** [examples/mechanics/CMakeFiles/FragmentingCylinder.dir/all] Error 2
make: *** [all] Error 2

Reply:
You can't call that inside the kernel - move it before and you can still use it inside

Reply:
Done